### ISSUE TITLE
Feature: Ignore index signature while sorting types

### ISSUE THREAD

**Issue by ntnyq** [azat-io/eslint-plugin-perfectionist#395]
### What rule do you want to change?

sort-object-types, sort-interfaces

### Describe the problem

I prefer the order like this:

- required 
- optional
- index signature

but the plugin treats index signatures as a required property.

Maybe we can add an options to ignore index signature.

Or is there any options can already make this happen?

### Code example

```ts
export type Foobar = {
  foo: string
  bar?: string
  [key: string]: any
}
```

Config is rule `perfectionist/sort-object-types` with `groupKind: 'required-first'`.

### Validations

- [X] Read the [docs](https://perfectionist.dev).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by hugop95** [comment]
@azat-io @OlivierZal 
I personally think that the `groupKind` option can be migrated to a system similar to what exists in `sort-classes` and `sort-modules`, with
- selectors: `multiline`, `method`, `index-signature`.
- modifiers: `optional`, `required`.

This would allow users to:
- Reproduce the current behavior with `required-first` / `optional-first` with `['required', 'optional', 'unknown']` for example.
- Be flexible enough to handle config like shown in this issue with `['required', 'optional', 'index-signature']`.

The disadvantage is that it does make the config a bit more verbose: it's easier to write `groupKind: 'required-first'` than a `groups` configuration, but it limits possibilities.

Let me know what you think!

**Comment by OlivierZal** [comment]
@hugop95, I would be in favor to migrate to the sort-classes / sort-modules system whenever it's possible: if you remember, at the beginning I proposed to use `groupKind: optional-first/required-first` for classes, but the system which was chosen is both flexible and transparent. It might be overkill for some given the small number of options for types/interfaces, but the resulted config shouldn't be more verbose though.

**Comment by azat-io** [comment]
Implemented in 8407bae.
Released in [v4.2.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v4.2.0).
